src/main/java/org/spongepowered/api/command/args/GenericArguments.java

```diff
@@ -1585,4 +1604,247 @@ public Text getUsage(CommandSource src) {
+            String s = args.next();
+            try {
+                return InetAddress.getByName(s);
+            } catch (UnknownHostException e) {
+                if (this.self) {
+                    if (source instanceof RemoteSource) {
+                        args.setState(state);
+                        return ((RemoteSource) source).getConnection().getAddress().getAddress();
+                    } else {
+                        throw args.createError(Text.of("Invalid IP address, and source was not a player!"));
+                    }
+                }
+                throw args.createError(Text.of("Invalid IP address!"));
+            }
+        }
+
+        @Override
+        public Text getUsage(CommandSource src) {
+            return src instanceof RemoteSource && this.self ? Text.of("[", super.getUsage(src), "]") : super.getUsage(src);
+        }
+
+    }
+
+    /**
+     * Expect an argument to return a {@link BigDecimal}.
+     *
+     * @param key The key to store under
+     * @return the argument
+     */
+    public static CommandElement bigDecimal(Text key) {
+        return new BigDecimalElement(key);
+    }
+
+    private static class BigDecimalElement extends KeyElement {
+
+        protected BigDecimalElement(Text key) {
+            super(key);
+        }
+
+        @Nullable
+        @Override
+        protected Object parseValue(CommandSource source, CommandArgs args) throws ArgumentParseException {
+            String next = args.next();
+            try {
+                return new BigDecimal(next);
+            } catch (NumberFormatException ex) {
+                throw args.createError(Text.of("Expected a number, but input " + next + " was not"));
+            }
+        }
+    }
+
+    /**
+     * Expect an argument to return a {@link BigInteger}.
+     *
+     * @param key The key to store under
+     * @return the argument
+     */
+    public static CommandElement bigInteger(Text key) {
+        return new BigIntegerElement(key);
+    }
+
+    private static class BigIntegerElement extends KeyElement {
+
+        protected BigIntegerElement(Text key) {
+            super(key);
+        }
+
+        @Nullable
+        @Override
+        protected Object parseValue(CommandSource source, CommandArgs args) throws ArgumentParseException {
+            String integerString = args.next();
+            try {
+                return new BigInteger(integerString);
+            } catch (NumberFormatException ex) {
+                throw args.createError(Text.of("Expected an integer, but input "+integerString+" was not"));
+            }
+        }
+    }
+
+    /**
+     * Expect an argument to be a valid {@link ConfigurationNode}.
+     *
+     * @param key The key to store under
+     * @param forceJson If true, requires JSON; if false, allows the more
+     *                  lenient HOCON
+     * @return the argument
+     */
+    public static CommandElement configNode(Text key, boolean forceJson) {
```

Comment:
That is a good point - it's not exactly clear that it's a 'config object'. I'll think of a new name.